src/web_socket.ts

```diff
@@ -96,64 +95,68 @@ export default class WebSocket extends EventEmitter {
   /**
    * Reconnects to the same endpoint.
    */
-  private _reconnect(cli: client) {
-    setTimeout(() => {
-      if (this._reconnectCurrentAttempts < this._reconnectMaxAttempts) {
-        this._reconnectCurrentAttempts++
-        // Call connect methods
-        console.log('Reconnecting')
-        this._connect(cli, this.url, this.stream, this._accessToken, this.headers)
-      }
-    }, this._reconnectInterval)
+  private _reconnect() {
+    if (this._client) {
+      setTimeout(() => {
+        if (this._reconnectCurrentAttempts < this._reconnectMaxAttempts) {
+          this._reconnectCurrentAttempts++
+          // Call connect methods
+          console.log('Reconnecting')
+          this._client = this._connect(this.url, this.stream, this._accessToken, this.headers)
+          this._bindSocket(this._client)
+        }
+      }, this._reconnectInterval)
+    }
   }
 
-  private _connect(cli: client, url: string, stream: string, accessToken: string, headers: {}) {
+  /**
+   * @param url Base url of streaming endpoint.
+   * @param stream The specified stream name.
+   * @param accessToken Access token.
+   * @param headers The specified headers.
+   * @return A WebSocket instance.
+   */
+  private _connect(url: string, stream: string, accessToken: string, headers: { [key: string]: string }): ws {
     const params: Array<string> = [`stream=${stream}`]
 
     if (accessToken !== null) {
       params.push(`access_token=${accessToken}`)
     }
-    const req_url: string = `${url}/?${params.join('&')}`
-    cli.connect(req_url, undefined, undefined, headers)
+    const requestURL: string = `${url}/?${params.join('&')}`
+    const options: ws.ClientOptions = {
+      headers: headers
+    }
+
+    const cli: ws = new ws(requestURL, options)
```

Comment:
__[ESLint]__ Error - A constructor name should not start with a lowercase letter. (new-cap) ([link](https://sider.review/gh/repos/136549493/pulls/52#issue-4308404))

<sub>posted by Sider</sub>